libcudacxx/include/cuda/__memory_resource/resource_ref.h

```diff
@@ -347,43 +365,59 @@ class basic_resource_ref
       , __vtable(__ref)
   {}
 
+  //! @brief Equality comparison between two \c basic_resource_ref
+  //! @param __rhs The other \c basic_resource_ref
+  //!
+  //! @return Checks whether both resources have the same equality function stored in their vtable and if so returns
+  //! that equality comparison. Otherwise returns false.
   _LIBCUDACXX_TEMPLATE(class... _OtherProperties)
   _LIBCUDACXX_REQUIRES((sizeof...(_Properties) == sizeof...(_OtherProperties))
                          _LIBCUDACXX_AND __properties_match<_OtherProperties...>)
-  _CCCL_NODISCARD bool operator==(const basic_resource_ref<_Alloc_type, _OtherProperties...>& __right) const
+  _CCCL_NODISCARD bool operator==(const basic_resource_ref<_Alloc_type, _OtherProperties...>& __rhs) const
   {
-    return (this->__static_vtable->__equal_fn == __right.__static_vtable->__equal_fn) //
-        && this->__static_vtable->__equal_fn(this->__object, __right.__object);
+    return (this->__static_vtable->__equal_fn == __rhs.__static_vtable->__equal_fn)
+        && this->__static_vtable->__equal_fn(this->__object, __rhs.__object);
   }
 
+  //! @brief Inequality comparison between two \c basic_resource_ref
+  //! @param __rhs The other \c basic_resource_ref
+  //!
+  //! @return Checks whether both resources have the same equality function stored in their vtable and if so returns
+  //! that inverse of the equality comparison. Otherwise returns true.
```

Comment:
```suggestion
  //! @return Checks whether both resources have the same equality function stored in their vtable and if so returns
  //! the inverse result of that equality comparison. Otherwise returns true.
```